### PR TITLE
Algodaily - day 02 - Array Intersection

### PR DESCRIPTION
**Pull request by Shernz**
Prompt for day 02 was finding the intersection of given 2 arrays

### CODE REVIEW

File: src/main/java/com/algodaily/technicalInterviewPrep/ArrayIntersection.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * Expected space complexity: O(max(n,m)).
+ */
+public class ArrayIntersection {
+    public static Integer[] arrayIntersection(int[] arr1, int[] arr2) {
+        // fill in
+        Set<Integer> intersection = new HashSet<>();
+        Set<Integer> result = new HashSet<>();
+
+        for(int num : arr1)
+            intersection.add(num);
```
Comment (Xnkr) [inline src/main/java/com/algodaily/technicalInterviewPrep/ArrayIntersection.java:39]: Could you be done at initialization

`Set<T> mySet = new HashSet<>(Arrays.asList(someArray));`

`Arrays.stream(strArray).collect(Collectors.toSet());`

`Set<String> strSet2 = Arrays.stream(stringArray).collect(Collectors.toCollection(HashSet::new));`

File: src/main/java/com/algodaily/technicalInterviewPrep/ArrayIntersection.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        for(int num : arr2)
+            if (intersection.contains(num))
+                result.add(num);
```
Comment (Xnkr) [inline src/main/java/com/algodaily/technicalInterviewPrep/ArrayIntersection.java:43]: Get into the habit of using streams

`Set<String> intersection = set1.stream()
    .filter(set2::contains)
    .collect(Collectors.toSet());`

Comment (Xnkr) [reply src/main/java/com/algodaily/technicalInterviewPrep/ArrayIntersection.java:43]: https://docs.oracle.com/javase/6/docs/api/java/util/Set.html#retainAll(java.util.Collection)

**Review by Xnkr — APPROVED** [review]

File: src/main/java/com/algodaily/technicalInterviewPrep/ArrayIntersection.java (hunk 1 above)
Comment (Shernz) [reply src/main/java/com/algodaily/technicalInterviewPrep/ArrayIntersection.java:39]: okay

File: src/main/java/com/algodaily/technicalInterviewPrep/ArrayIntersection.java (hunk 2 above)
Comment (Shernz) [reply src/main/java/com/algodaily/technicalInterviewPrep/ArrayIntersection.java:43]: i tried many times to use streams but i get some or the other syntax error

**Pull request merged**